### PR TITLE
[YUNIKORN-2623] Create unit tests for Clients

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
Create mock for `SharedIndexedInformer` and update existing types.
Write new unit tests for `client.Clients`.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2623

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/client/clients_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+	"github.com/apache/yunikorn-core/pkg/common"
+	"github.com/apache/yunikorn-k8shim/pkg/common/test"
+)
+
+const (
+	noOfInformers = 9
+)
+
+func TestWaitForSync(t *testing.T) {
```
Comment (chia7712) [inline pkg/client/clients_test.go:35]: Maybe we can do a bit refactor to `Clients` to simplify the test. 

1. add a new method `informers` to  `Clients` to return all informers
2. rewrite `WaitForSync` and `Run` by looping all informers
3. in this test we can create a mock clients to override the new method `informers` to return a single mock informer

WDYT?

Comment (pbacsko) [reply pkg/client/clients_test.go:35]: Could we do it a separate ticket? In this one I'd rather just focus on the new tests.

Comment (chia7712) [reply pkg/client/clients_test.go:35]: ok, I have filed https://issues.apache.org/jira/browse/YUNIKORN-2625

File: pkg/client/apifactory_mock.go [hunk 2]
```diff
@@ -416,38 +417,80 @@ func (m *MockedAPIProvider) GetBoundPods(clear bool) []BoundPod {
...
+}
+
+func NewMockedCSINodeInformer() *MockedCSINodeInformer {
+	return &MockedCSINodeInformer{
+		informer: &test.SharedInformerMock{},
+	}
+}
+
+func (m *MockedCSINodeInformer) Informer() cache.SharedIndexInformer {
+	return &test.SharedInformerMock{}
```
Comment (chia7712) [inline pkg/client/apifactory_mock.go:487]: maybe it should return `m.informer`?

File: pkg/client/clients_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	"time"
+
+	"gotest.tools/v3/assert"
+
+	"github.com/apache/yunikorn-core/pkg/common"
+	"github.com/apache/yunikorn-k8shim/pkg/common/test"
+)
+
+const (
+	noOfInformers = 9
```
Comment (chia7712) [inline pkg/client/clients_test.go:32]: Could you add comments for this magic number?

**Review by chia7712 — COMMENTED** [review]
@pbacsko thanks for this patch!

File: pkg/client/apifactory_mock.go (hunk 2 above)
Comment (pbacsko) [reply pkg/client/apifactory_mock.go]: Correct, fixed.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request closed without merging**